sauce/main.js

```diff
@@ -27,7 +27,19 @@ const featureInstances = features.map(Feature => new Feature());
       if (feature.settings.enabled) {
         feature.applyListeners();
 
-        if (feature.shouldInvoke()) feature.invoke();
+        const willInvokeRetValue = feature.willInvoke();
+
+        if (willInvokeRetValue && typeof willInvokeRetValue.then === 'function') {
+          willInvokeRetValue.then(() => {
```

Comment:
Theoretically, could we do this?

```javascript
await feature.willInvoke();

if (feature.shouldInvoke()) {
  feature.invoke();
}
```

I'm successfully transpiling `async`/`await` at work with great happiness, and I reckon we could do the same here. If babel doesn't play nice I'll find my config at work.

Would save quite a bit of malarky I reckon.